src/framework/jobState/LocalStepJobState/LocalStepJobState.ts

```diff
@@ -0,0 +1,97 @@
+import { Promisable } from 'type-fest';
+
+import { Entity, Relationship } from '../../types';
+import { JobState, GraphObjectFilter, GraphObjectIteratee } from '../types';
+
+import { flushDataToDisk } from './flushDataToDisk';
+
+export const GRAPH_OBJECT_BUFFER_THRESHOLD = 500; // arbitrarily selected, subject to tuning
+
+interface LocalStepJobStateInput {
+  step: string;
+  cacheDirectory?: string;
```

Comment:
As I think of this as a `FileSystemJobState`, I now wonder if the name here should just be `rootDirectory`. It isn't clear to me (from the `FileSystemJobState` perspective I suppose), what would be "cached", whereas `rootDirectory` clearly communicates the location in which all content will be maintained and it isn't really a cache. Or is it?

Reply:
I suppose `cache` is probably the wrong terminology since the data isn't really hidden (although by default data is  `.j1-integration` directory, which suggests that it's a hidden directory. I can change this for the whole project in a later PR since the terminology is use all over the place.

Reply:
Ahh, that's helpful for understanding where the name came from, thank you! FWIW, I don't think of being hidden an intrinsic property of a cache. I tend to call something a cache if it's meant to keep expensive work from happening, and a buffer if it's meant to delay future work.